src/daemon.py

```diff
@@ -199,6 +203,19 @@ def write_coingecko_prices(
         logger.error("Error inserting record: %s", e)
 
 
+def get_price(
+    PriceProviders: List, block_number: int, token_address: str
+) -> Optional[Tuple[float, str]]:
+    for provider in PriceProviders:
+        try:
+            price = provider.get_price(block_number, token_address)
```

Comment:
If you use List from the typing library you can set the types within the list statically as well: `PriceProviders: List[Provider]`.